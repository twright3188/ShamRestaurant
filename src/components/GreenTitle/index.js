import React from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet } from 'react-native';
import Colors from '../../theme';

const styles = StyleSheet.create({
  captionText: {
    position: 'relative',
    width: '100%',
    color: 'black',
    
    fontFamily: Colors.FontBaseBold,
    fontWeight: 'bold',
  }
});

function GreenTitle(props) {
  const { children, fontSize, style } = props;

  return (
    <Text style={[styles.captionText, { fontSize, lineHeight: fontSize * 1.2 }, style]}>{children}</Text>
  );
}

GreenTitle.propTypes = {
  children: PropTypes.node,
  fontSize: PropTypes.number,
  style: PropTypes.objectOf(PropTypes.object)
};

GreenTitle.defaultProps = {
  children: '',
  fontSize: 20,
  style: {}
};

export default GreenTitle;