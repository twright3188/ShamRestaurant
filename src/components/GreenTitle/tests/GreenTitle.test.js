import React from 'react';
import renderer from 'react-test-renderer';
import GreenTitle  from '../index';

it ('green title renders correctly', () => {
    const tree = renderer.create(<GreenTitle>test</GreenTitle>).toJSON();
    expect(tree).toMatchSnapshot();
});