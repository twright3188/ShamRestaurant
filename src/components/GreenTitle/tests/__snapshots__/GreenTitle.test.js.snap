// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`green title renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "alignItems": "center",
        "color": "#399AE5",
        "fontFamily": "Helvetica Neue",
        "fontWeight": "bold",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      },
      Object {
        "fontSize": 20,
        "lineHeight": 24,
      },
      Object {},
    ]
  }
>
  test
</Text>
`;
